Code/Tools/FBuild/FBuildCore/Graph/NodeGraph.cpp

```diff
@@ -1436,6 +1436,77 @@ Node * NodeGraph::FindNodeInternal( const AString & fullPath ) const
 	return nullptr;
 }
 
+// FindNearestNodesInternal
+//------------------------------------------------------------------------------
+size_t NodeGraph::FindNearestNodesInternal( const AString & fullPath, const Node **nodes, uint32_t capacity ) const
+{
+	ASSERT( Thread::IsMainThread() );
+	ASSERT( nodes );
+	ASSERT( capacity );
+
+	if ( fullPath.IsEmpty() )
+		return 0;
+
+	const uint32_t maxDistance = 5;
+	uint32_t worstMinDistance = fullPath.GetLength() + 1;
+
+	ASSERT( capacity <= 8 );
+	uint32_t nodesDistance[8] = {0};
+
+	size_t count = 0;
+	for ( size_t i = 0 ; i < NODEMAP_TABLE_SIZE ; i++ )
+	{
+		if ( nullptr == m_NodeMap[i] )
+			continue;
+
+		Node * const n = m_NodeMap[i];
```

Comment:
The m_NodeMap is a table of linked lists (nodes with colliding hashes are chained together). You need to iterate the linked list to make sure you get all the nodes.